src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -145,8 +151,8 @@ def generate_json_report(self):
     def _populate_core_stats(self, area):
         if area.uuid not in self.flattened_area_core_stats_dict:
             self.flattened_area_core_stats_dict[area.uuid] = {}
-        if self.current_market_unix is not None and \
-                self.current_market_unix not in self.flattened_area_core_stats_dict[area.uuid]:
+        if self.current_market != "" and \
+                self.current_market not in self.flattened_area_core_stats_dict[area.uuid]:
```

Comment:
So you have decided to move from unix time to time string again. Any particular reason for that?

Reply:
Yes, wanted to start with following unix_time_stamp but realised would be needing a lot of reconversion in between, so postponed and to be dealt carefully as part of `1739`

Reply:
ok, I see now. The d3a work for `1739` is done btw. Have a look here:
https://github.com/gridsingularity/d3a/blob/02068b2ca1c125626e8765afd144c9883c825e64/src/d3a/d3a_core/sim_results/endpoint_buffer.py#L117
Where the `convert_pendulum_to_str_in_dict` is used. When we tackle https://gridsingularity.atlassian.net/browse/D3ASIM-2288, we will enable the `unix_time=True` and convert it in d3a-web.
So, yes, you are right, no need to use this unix_time here.

Reply:
Yes, but my idea is to scratch the datetime format and string formatted datetime completely from d3a and Im sure it will make the simulation optimized at least in terms of memory usage